cmd/cloudflare.go

```diff
@@ -0,0 +1,80 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/konstructio/dropkick/internal/cloudflare"
+	"github.com/konstructio/dropkick/internal/logger"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type cloudflareOptions struct {
+	nuke      bool
+	domain    string
+	subdomain string
+	quiet     bool
+}
+
+func getCloudflareCommand() *cobra.Command {
+	var opts cloudflareOptions
+
+	cloudflareCmd := &cobra.Command{
+		Use:   "cloudflare",
+		Short: "clean cloudflare dns resources",
+		Long:  `clean cloudflare dns resources`,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			opts.quiet = cmd.Flags().Lookup("quiet").Value.String() == "true"
+			return runCloudflare(cmd.Context(), cmd.OutOrStderr(), opts, os.Getenv("CLOUDFLARE_API_TOKEN"))
+		},
+	}
+
+	cloudflareCmd.Flags().BoolVar(&opts.nuke, "nuke", false, "required to confirm deletion of resources")
+	cloudflareCmd.Flags().StringVar(&opts.domain, "domain", "", "the cloudflare apex domain to clean")
+	cloudflareCmd.Flags().StringVar(&opts.subdomain, "subdomain", "", "the subdomain to clean")
+
+	if err := cloudflareCmd.MarkFlagRequired("domain"); err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Marking a nonexistent flag as required will fail as soon as the CLI starts (potentially with a panic too, haven't checked), so this could be simplified to:

```suggestion
	cloudflareCmd.MarkFlagRequired("domain")
```